src/main/java/me/mrCookieSlime/Slimefun/api/inventory/BlockMenu.java

```diff
@@ -53,19 +54,34 @@ public void save(Location l) {
             return;
         }
 
+        Block b = l.getBlock();
+        if (b == null) {
+            return;
+        }
+
         // To force CS-CoreLib to build the Inventory
         this.getContents();
 
-        File file = new File("data-storage/Slimefun/stored-inventories/" + serializeLocation(l) + ".sfi");
-        Config cfg = new Config(file);
-        cfg.setValue("preset", preset.getID());
+        Plugin plugin = Slimefun.getPlugin(Slimefun.class);
+        CustomBlockData blockData = new CustomBlockData(b, plugin);
+        blockData.set(new NamespacedKey(plugin, "preset"), PersistentDataType.STRING, preset.getID());
```

Comment:
Convert to use BlockStorage.getLocationInfo instead of directly accessing this in such a horrible way